packages/repository/src/model.ts

```diff
@@ -358,10 +359,11 @@ export abstract class Model {
     const def = (this.constructor as typeof Model).definition;
     const obj: AnyObject = {};
 
+    const props = def?.properties;
     if (options?.ignoreUnknownProperties === false) {
       const hiddenProperties: string[] = def?.settings.hiddenProperties || [];
       for (const p in this) {
-        if (!hiddenProperties.includes(p)) {
+        if (!hiddenProperties.includes(p) && !props?.[p]?.hidden) {
```

Comment:
Can we revert this change please? I believe we agreed to handle `hidden` flag in the model definition only.

It would be best to revert all changes in `toJSON` method.

Reply:
yes of course